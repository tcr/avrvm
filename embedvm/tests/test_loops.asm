PushZeros(n=1)
PushConstantV(value=0)
PopLocal(sfa=0)
Label(id='label4')
PushLocal(sfa=0)
PushConstantV(value=5)
CompareLT()
JumpVIfNot(address=LabelRef('label1'))
PushLocal(sfa=0)
PushConstantV(value=3)
CompareEE()
JumpVIfNot(address=LabelRef('label2'))
JumpV(address=LabelRef('label3'))
Label(id='label2')
PushLocal(sfa=0)
PushConstantV(value=1)
CallUserFunction(funcid=1)
DropValue()
Label(id='label3')
PushLocal(sfa=0)
PushConstantV(value=1)
Add()
PopLocal(sfa=0)
JumpV(address=LabelRef('label4'))
Label(id='label1')
PushLocal(sfa=0)
PushConstantV(value=1)
CallUserFunction(funcid=2)
DropValue()
PushLocal(sfa=0)
PushConstantV(value=1)
Sub()
Bury(k=0)
PopLocal(sfa=0)
PushConstantV(value=0)
CompareGT()
JumpVIf(address=LabelRef('label1'))
PushLocal(sfa=0)
PopLocal(sfa=1)
Label(id='label8')
PushLocal(sfa=0)
PushConstantV(value=20)
CompareLT()
JumpVIfNot(address=LabelRef('label6'))
PushLocal(sfa=0)
PushConstantV(value=1)
Add()
PopLocal(sfa=0)
PushLocal(sfa=0)
PushConstantV(value=2)
Mod()
PushConstantV(value=0)
CompareEE()
JumpVIfNot(address=LabelRef('label7'))
JumpV(address=LabelRef('label8'))
Label(id='label7')
PushLocal(sfa=1)
Bury(k=0)
PushConstantV(value=1)
Add()
PopLocal(sfa=1)
PushLocal(sfa=0)
PushConstantV(value=2)
CallUserFunction(funcid=3)
DropValue()
JumpV(address=LabelRef('label8'))
Label(id='label6')
PushLocal(sfa=0)
Bury(k=0)
PushConstantV(value=1)
Add()
PopLocal(sfa=0)
PushConstantV(value=50)
CompareLT()
JumpVIfNot(address=LabelRef('label10'))
PushLocal(sfa=0)
PushConstantV(value=23)
CompareEE()
JumpVIfNot(address=LabelRef('label11'))
JumpV(address=LabelRef('label10'))
Label(id='label11')
PushLocal(sfa=0)
PushConstantV(value=1)
CallUserFunction(funcid=4)
DropValue()
JumpV(address=LabelRef('label6'))
Label(id='label10')
PushLocal(sfa=0)
PushConstantV(value=3)
Sub()
PopLocal(sfa=1)
Label(id='label18')
PushLocal(sfa=0)
PushLocal(sfa=1)
Add()
PushLocal(sfa=1)
PushConstantV(value=2)
CallUserFunction(funcid=5)
DropValue()
PushLocal(sfa=0)
PushLocal(sfa=1)
Add()
PushConstantV(value=50)
CompareEE()
JumpVIfNot(address=LabelRef('label14'))
JumpV(address=LabelRef('label15'))
Label(id='label14')
PushLocal(sfa=0)
PushLocal(sfa=1)
Add()
PushConstantV(value=2)
Mod()
PushConstantV(value=0)
CompareEE()
JumpVIfNot(address=LabelRef('label16'))
JumpV(address=LabelRef('label17'))
Label(id='label16')
PushLocal(sfa=0)
PushLocal(sfa=1)
Add()
PushConstantV(value=1)
CallUserFunction(funcid=6)
DropValue()
Label(id='label17')
PushLocal(sfa=1)
PushConstantV(value=1)
Add()
PopLocal(sfa=1)
JumpV(address=LabelRef('label18'))
Label(id='label15')
PushConstantV(value=0)
PopLocal(sfa=0)
Label(id='label23')
PushLocal(sfa=0)
PushConstantV(value=5)
CompareLT()
JumpVIfNot(address=LabelRef('label19'))
PushLocal(sfa=0)
PushConstantV(value=2)
Mod()
PushConstantV(value=0)
CompareNE()
JumpVIfNot(address=LabelRef('label20'))
JumpV(address=LabelRef('label21'))
Label(id='label20')
PushLocal(sfa=0)
PushConstantV(value=0)
PushConstantV(value=2)
CallUserFunction(funcid=7)
DropValue()
PushConstantV(value=0)
JumpVIfNot(address=LabelRef('label22'))
Label(id='label21')
PushLocal(sfa=0)
PushConstantV(value=1)
PushConstantV(value=2)
CallUserFunction(funcid=7)
DropValue()
Label(id='label22')
PushLocal(sfa=0)
PushConstantV(value=1)
Add()
PopLocal(sfa=0)
JumpV(address=LabelRef('label23'))
Label(id='label19')
Return0()