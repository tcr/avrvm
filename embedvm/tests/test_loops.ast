 LEFT 0x7fd88b407d90 @ 0000 0000:
    LEFT 0x7fd88b407c90 @ 0000 0000: op=f1
   LEFT 0x7fd88b407cd0 @ 0000 0001:
          LEFT 0x7fd88b404350 @ 0001 0001:
             LEFT 0x7fd88b403ae0 @ 0001 0001: op=90
            LEFT 0x7fd88b403b20 @ 0001 0002:
             RIGHT 0x7fd88b403aa0 @ 0002 0002: op=40
           RIGHT 0x7fd88b404310 @ 0001 0003:
               LEFT 0x7fd88b403b70 @ 0003 0003: op=00
              LEFT 0x7fd88b403bf0 @ 0003 0004:
               RIGHT 0x7fd88b403bb0 @ 0004 0004: op=98 arg=05
             LEFT 0x7fd88b403c30 @ 0003 0006: op=a8
            RIGHT 0x7fd88b4042d0 @ 0003 0007: op=a6 arg=13 rel argaddr=0x7fd88b403de0
                LEFT 0x7fd88b404070 @ 0009 0009:
                    LEFT 0x7fd88b403e70 @ 0009 0009: op=00
                   LEFT 0x7fd88b403ef0 @ 0009 000a:
                    RIGHT 0x7fd88b403eb0 @ 000a 000a: op=93
                  LEFT 0x7fd88b403f30 @ 0009 000b: op=aa
                 RIGHT 0x7fd88b404030 @ 0009 000c: op=a6 arg=04 rel argaddr=0x7fd88b403fb0
                   LEFT 0x7fd88b403f70 @ 000e 000e: op=a0 arg=06 rel argaddr=0x7fd88b403e20
                  RIGHT 0x7fd88b403ff0 @ 000e 0010:
                   RIGHT 0x7fd88b403fb0 @ 0010 0010:
               LEFT 0x7fd88b4041d0 @ 0009 0010:
                   LEFT 0x7fd88b4040c0 @ 0010 0010: op=00
                  LEFT 0x7fd88b404110 @ 0010 0011: op=91
                 LEFT 0x7fd88b404150 @ 0010 0012: op=b1
                RIGHT 0x7fd88b404190 @ 0010 0013: op=9d
              LEFT 0x7fd88b404250 @ 0009 0014:
                LEFT 0x7fd88b403e20 @ 0014 0014:
               RIGHT 0x7fd88b404210 @ 0014 0014:
                  LEFT 0x7fd88b403c80 @ 0014 0014: op=00
                 LEFT 0x7fd88b403d40 @ 0014 0015:
                   LEFT 0x7fd88b403cc0 @ 0015 0015: op=91
                  RIGHT 0x7fd88b403d00 @ 0015 0016: op=80
                RIGHT 0x7fd88b403d80 @ 0014 0017: op=40
             RIGHT 0x7fd88b404290 @ 0009 0018: op=a0 arg=eb rel argaddr=0x7fd88b403c30
              RIGHT 0x7fd88b403de0 @ 001a 001a:
         LEFT 0x7fd88b4048e0 @ 0001 001a:
              LEFT 0x7fd88b404550 @ 001a 001a:
                  LEFT 0x7fd88b404440 @ 001a 001a: op=00
                 LEFT 0x7fd88b404490 @ 001a 001b: op=91
                LEFT 0x7fd88b4044d0 @ 001a 001c: op=b2
               RIGHT 0x7fd88b404510 @ 001a 001d: op=9d
             LEFT 0x7fd88b4047e0 @ 001a 001e:
              RIGHT 0x7fd88b4043f0 @ 001e 001e:
            LEFT 0x7fd88b404820 @ 001a 001e:
                  LEFT 0x7fd88b4045a0 @ 001e 001e: op=00
                 LEFT 0x7fd88b404660 @ 001e 001f:
                   LEFT 0x7fd88b4045e0 @ 001f 001f: op=91
                  RIGHT 0x7fd88b404620 @ 001f 0020: op=81
                LEFT 0x7fd88b4046a0 @ 001e 0021: op=c5
               LEFT 0x7fd88b4046e0 @ 001e 0022: op=40
              LEFT 0x7fd88b404760 @ 001e 0023:
               RIGHT 0x7fd88b404720 @ 0023 0023: op=90
             RIGHT 0x7fd88b4047a0 @ 001e 0024: op=ad
           LEFT 0x7fd88b404860 @ 001a 0025: op=a4 arg=f5 rel argaddr=0x7fd88b4047e0
          RIGHT 0x7fd88b4048a0 @ 001a 0027:
           RIGHT 0x7fd88b4043b0 @ 0027 0027:
        LEFT 0x7fd88b404a00 @ 0001 0027:
          LEFT 0x7fd88b404980 @ 0027 0027: op=00
         RIGHT 0x7fd88b4049c0 @ 0027 0028:
          RIGHT 0x7fd88b404930 @ 0028 0028: op=41
       LEFT 0x7fd88b405500 @ 0001 0029:
          LEFT 0x7fd88b404bb0 @ 0029 0029:
         LEFT 0x7fd88b405480 @ 0029 0029:
             LEFT 0x7fd88b404a50 @ 0029 0029: op=00
            LEFT 0x7fd88b404ad0 @ 0029 002a:
             RIGHT 0x7fd88b404a90 @ 002a 002a: op=98 arg=14
           LEFT 0x7fd88b404b10 @ 0029 002c: op=a8
          RIGHT 0x7fd88b405440 @ 0029 002d: op=a6 arg=1a rel argaddr=0x7fd88b4053c0
              LEFT 0x7fd88b404d40 @ 002f 002f:
                 LEFT 0x7fd88b404c00 @ 002f 002f: op=00
                LEFT 0x7fd88b404cc0 @ 002f 0030:
                  LEFT 0x7fd88b404c40 @ 0030 0030: op=91
                 RIGHT 0x7fd88b404c80 @ 0030 0031: op=80
               RIGHT 0x7fd88b404d00 @ 002f 0032: op=40
             LEFT 0x7fd88b405050 @ 002f 0033:
                   LEFT 0x7fd88b404d90 @ 0033 0033: op=00
                  LEFT 0x7fd88b404e10 @ 0033 0034:
                   RIGHT 0x7fd88b404dd0 @ 0034 0034: op=92
                 LEFT 0x7fd88b404e50 @ 0033 0035: op=84
                LEFT 0x7fd88b404ed0 @ 0033 0036:
                 RIGHT 0x7fd88b404e90 @ 0036 0036: op=90
               LEFT 0x7fd88b404f10 @ 0033 0037: op=aa
              RIGHT 0x7fd88b405010 @ 0033 0038: op=a6 arg=04 rel argaddr=0x7fd88b404f90
                LEFT 0x7fd88b404f50 @ 003a 003a: op=a0 arg=ef rel argaddr=0x7fd88b404bb0
               RIGHT 0x7fd88b404fd0 @ 003a 003c:
                RIGHT 0x7fd88b404f90 @ 003c 003c:
            LEFT 0x7fd88b405380 @ 002f 003c:
                    LEFT 0x7fd88b405100 @ 003c 003c: op=01
                   LEFT 0x7fd88b4051c0 @ 003c 003d: op=c5
                  LEFT 0x7fd88b405200 @ 003c 003e:
                    LEFT 0x7fd88b405140 @ 003e 003e: op=91
                   RIGHT 0x7fd88b405180 @ 003e 003f: op=80
                 LEFT 0x7fd88b405240 @ 003c 0040: op=41
                LEFT 0x7fd88b405280 @ 003c 0041:
                 RIGHT 0x7fd88b4050a0 @ 0041 0041: op=00
               LEFT 0x7fd88b4052c0 @ 003c 0042: op=92
              LEFT 0x7fd88b405300 @ 003c 0043: op=b3
             RIGHT 0x7fd88b405340 @ 003c 0044: op=9d
           RIGHT 0x7fd88b405400 @ 002f 0045: op=a0 arg=e4 rel argaddr=0x7fd88b404b10
            RIGHT 0x7fd88b4053c0 @ 0047 0047:
        RIGHT 0x7fd88b4054c0 @ 0029 0047:
         RIGHT 0x7fd88b404b70 @ 0047 0047:
      LEFT 0x7fd88b405d20 @ 0001 0047:
         LEFT 0x7fd88b4057f0 @ 0047 0047:
        LEFT 0x7fd88b405ca0 @ 0047 0047:
               LEFT 0x7fd88b405550 @ 0047 0047: op=00
              LEFT 0x7fd88b405610 @ 0047 0048: op=c5
             LEFT 0x7fd88b405650 @ 0047 0049:
               LEFT 0x7fd88b405590 @ 0049 0049: op=91
              RIGHT 0x7fd88b4055d0 @ 0049 004a: op=80
            LEFT 0x7fd88b405690 @ 0047 004b: op=40
           LEFT 0x7fd88b405710 @ 0047 004c:
            RIGHT 0x7fd88b4056d0 @ 004c 004c: op=98 arg=32
          LEFT 0x7fd88b405750 @ 0047 004e: op=a8
         RIGHT 0x7fd88b405c60 @ 0047 004f: op=a6 arg=10 rel argaddr=0x7fd88b405be0
            LEFT 0x7fd88b405a40 @ 0051 0051:
                LEFT 0x7fd88b405840 @ 0051 0051: op=00
               LEFT 0x7fd88b4058c0 @ 0051 0052:
                RIGHT 0x7fd88b405880 @ 0052 0052: op=98 arg=17
              LEFT 0x7fd88b405900 @ 0051 0054: op=aa
             RIGHT 0x7fd88b405a00 @ 0051 0055: op=a6 arg=04 rel argaddr=0x7fd88b405980
               LEFT 0x7fd88b405940 @ 0057 0057: op=a0 arg=08 rel argaddr=0x7fd88b4057b0
              RIGHT 0x7fd88b4059c0 @ 0057 0059:
               RIGHT 0x7fd88b405980 @ 0059 0059:
           LEFT 0x7fd88b405ba0 @ 0051 0059:
               LEFT 0x7fd88b405a90 @ 0059 0059: op=00
              LEFT 0x7fd88b405ae0 @ 0059 005a: op=91
             LEFT 0x7fd88b405b20 @ 0059 005b: op=b4
            RIGHT 0x7fd88b405b60 @ 0059 005c: op=9d
          RIGHT 0x7fd88b405c20 @ 0051 005d: op=a0 arg=ea rel argaddr=0x7fd88b405750
           RIGHT 0x7fd88b405be0 @ 005f 005f:
       RIGHT 0x7fd88b405ce0 @ 0047 005f:
        RIGHT 0x7fd88b4057b0 @ 005f 005f:
     LEFT 0x7fd88b406e20 @ 0001 005f:
           LEFT 0x7fd88b405dc0 @ 005f 005f: op=00
          LEFT 0x7fd88b405e40 @ 005f 0060:
           RIGHT 0x7fd88b405e00 @ 0060 0060: op=93
         LEFT 0x7fd88b405e80 @ 005f 0061: op=81
        LEFT 0x7fd88b405ec0 @ 005f 0062:
         RIGHT 0x7fd88b405d70 @ 0062 0062: op=41
       LEFT 0x7fd88b406da0 @ 005f 0063:
             LEFT 0x7fd88b406370 @ 0063 0063:
                    LEFT 0x7fd88b406160 @ 0063 0063: op=00
                   LEFT 0x7fd88b4061f0 @ 0063 0064:
                    RIGHT 0x7fd88b4061b0 @ 0064 0064: op=01
                  LEFT 0x7fd88b406230 @ 0063 0065: op=80
                 LEFT 0x7fd88b406270 @ 0063 0066:
                  RIGHT 0x7fd88b406100 @ 0066 0066: op=01
                LEFT 0x7fd88b4062b0 @ 0063 0067: op=92
               LEFT 0x7fd88b4062f0 @ 0063 0068: op=b5
              RIGHT 0x7fd88b406330 @ 0063 0069: op=9d
            LEFT 0x7fd88b406690 @ 0063 006a:
                  LEFT 0x7fd88b4063c0 @ 006a 006a: op=00
                 LEFT 0x7fd88b406450 @ 006a 006b:
                  RIGHT 0x7fd88b406410 @ 006b 006b: op=01
                LEFT 0x7fd88b406490 @ 006a 006c: op=80
               LEFT 0x7fd88b406510 @ 006a 006d:
                RIGHT 0x7fd88b4064d0 @ 006d 006d: op=98 arg=32
              LEFT 0x7fd88b406550 @ 006a 006f: op=aa
             RIGHT 0x7fd88b406650 @ 006a 0070: op=a6 arg=04 rel argaddr=0x7fd88b4065d0
               LEFT 0x7fd88b406590 @ 0072 0072: op=a0 arg=19 rel argaddr=0x7fd88b406070
              RIGHT 0x7fd88b406610 @ 0072 0074:
               RIGHT 0x7fd88b4065d0 @ 0074 0074:
           LEFT 0x7fd88b406a70 @ 0063 0074:
                   LEFT 0x7fd88b4066e0 @ 0074 0074: op=00
                  LEFT 0x7fd88b406770 @ 0074 0075:
                   RIGHT 0x7fd88b406730 @ 0075 0075: op=01
                 LEFT 0x7fd88b4067b0 @ 0074 0076: op=80
                LEFT 0x7fd88b406830 @ 0074 0077:
                 RIGHT 0x7fd88b4067f0 @ 0077 0077: op=92
               LEFT 0x7fd88b406870 @ 0074 0078: op=84
              LEFT 0x7fd88b4068f0 @ 0074 0079:
               RIGHT 0x7fd88b4068b0 @ 0079 0079: op=90
             LEFT 0x7fd88b406930 @ 0074 007a: op=aa
            RIGHT 0x7fd88b406a30 @ 0074 007b: op=a6 arg=04 rel argaddr=0x7fd88b4069b0
              LEFT 0x7fd88b406970 @ 007d 007d: op=a0 arg=08 rel argaddr=0x7fd88b4060b0
             RIGHT 0x7fd88b4069f0 @ 007d 007f:
              RIGHT 0x7fd88b4069b0 @ 007f 007f:
          LEFT 0x7fd88b406ca0 @ 0063 007f:
                LEFT 0x7fd88b406ac0 @ 007f 007f: op=00
               LEFT 0x7fd88b406b50 @ 007f 0080:
                RIGHT 0x7fd88b406b10 @ 0080 0080: op=01
              LEFT 0x7fd88b406b90 @ 007f 0081: op=80
             LEFT 0x7fd88b406be0 @ 007f 0082: op=91
            LEFT 0x7fd88b406c20 @ 007f 0083: op=b6
           RIGHT 0x7fd88b406c60 @ 007f 0084: op=9d
         LEFT 0x7fd88b406d20 @ 0063 0085:
           LEFT 0x7fd88b4060b0 @ 0085 0085:
          RIGHT 0x7fd88b406ce0 @ 0085 0085:
             LEFT 0x7fd88b405f10 @ 0085 0085: op=01
            LEFT 0x7fd88b405fd0 @ 0085 0086:
              LEFT 0x7fd88b405f50 @ 0086 0086: op=91
             RIGHT 0x7fd88b405f90 @ 0086 0087: op=80
           RIGHT 0x7fd88b406010 @ 0085 0088: op=41
        RIGHT 0x7fd88b406d60 @ 0063 0089: op=a0 arg=da rel argaddr=0x7fd88b406ca0
      RIGHT 0x7fd88b406de0 @ 005f 008b:
       RIGHT 0x7fd88b406070 @ 008b 008b:
    RIGHT 0x7fd88b407c50 @ 0001 008b:
       LEFT 0x7fd88b406eb0 @ 008b 008b: op=90
      LEFT 0x7fd88b406ef0 @ 008b 008c:
       RIGHT 0x7fd88b406e70 @ 008c 008c: op=40
     RIGHT 0x7fd88b407c10 @ 008b 008d:
         LEFT 0x7fd88b406f40 @ 008d 008d: op=00
        LEFT 0x7fd88b406fc0 @ 008d 008e:
         RIGHT 0x7fd88b406f80 @ 008e 008e: op=98 arg=05
       LEFT 0x7fd88b407000 @ 008d 0090: op=a8
      RIGHT 0x7fd88b407bd0 @ 008d 0091: op=a6 arg=1e rel argaddr=0x7fd88b4071b0
           LEFT 0x7fd88b407580 @ 0093 0093:
                 LEFT 0x7fd88b407240 @ 0093 0093: op=00
                LEFT 0x7fd88b4072c0 @ 0093 0094:
                 RIGHT 0x7fd88b407280 @ 0094 0094: op=92
               LEFT 0x7fd88b407300 @ 0093 0095: op=84
              LEFT 0x7fd88b407380 @ 0093 0096:
               RIGHT 0x7fd88b407340 @ 0096 0096: op=90
             LEFT 0x7fd88b4073c0 @ 0093 0097: op=ab
            RIGHT 0x7fd88b407540 @ 0093 0098: op=a6 arg=04 rel argaddr=0x7fd88b4074c0
              LEFT 0x7fd88b407480 @ 009a 009a: op=a0 arg=0a rel argaddr=0x7fd88b407440
             RIGHT 0x7fd88b407500 @ 009a 009c:
              RIGHT 0x7fd88b4074c0 @ 009c 009c:
          LEFT 0x7fd88b407760 @ 0093 009c:
               LEFT 0x7fd88b407620 @ 009c 009c: op=00
              LEFT 0x7fd88b407660 @ 009c 009d:
               RIGHT 0x7fd88b4075c0 @ 009d 009d: op=90
             LEFT 0x7fd88b4076a0 @ 009c 009e: op=92
            LEFT 0x7fd88b4076e0 @ 009c 009f: op=b7
           RIGHT 0x7fd88b407720 @ 009c 00a0: op=9d
         LEFT 0x7fd88b407ad0 @ 0093 00a1:
           LEFT 0x7fd88b4077a0 @ 00a1 00a1: op=90
          RIGHT 0x7fd88b407a90 @ 00a1 00a2: op=a6 arg=07 rel argaddr=0x7fd88b407a10
            LEFT 0x7fd88b4079d0 @ 00a4 00a4:
              LEFT 0x7fd88b407440 @ 00a4 00a4:
             RIGHT 0x7fd88b407990 @ 00a4 00a4:
                  LEFT 0x7fd88b407850 @ 00a4 00a4: op=00
                 LEFT 0x7fd88b407890 @ 00a4 00a5:
                  RIGHT 0x7fd88b4077f0 @ 00a5 00a5: op=91
                LEFT 0x7fd88b4078d0 @ 00a4 00a6: op=92
               LEFT 0x7fd88b407910 @ 00a4 00a7: op=b7
              RIGHT 0x7fd88b407950 @ 00a4 00a8: op=9d
           RIGHT 0x7fd88b407a50 @ 00a4 00a9:
            RIGHT 0x7fd88b407a10 @ 00a9 00a9:
        LEFT 0x7fd88b407b50 @ 0093 00a9:
          LEFT 0x7fd88b4071f0 @ 00a9 00a9:
         RIGHT 0x7fd88b407b10 @ 00a9 00a9:
            LEFT 0x7fd88b407050 @ 00a9 00a9: op=00
           LEFT 0x7fd88b407110 @ 00a9 00aa:
             LEFT 0x7fd88b407090 @ 00aa 00aa: op=91
            RIGHT 0x7fd88b4070d0 @ 00aa 00ab: op=80
          RIGHT 0x7fd88b407150 @ 00a9 00ac: op=40
       RIGHT 0x7fd88b407b90 @ 0093 00ad: op=a0 arg=e0 rel argaddr=0x7fd88b407000
        RIGHT 0x7fd88b4071b0 @ 00af 00af:
  RIGHT 0x7fd88b407d10 @ 0000 00af: sym=main op=9c
ROOT 0x7fd88b407e20 @ 0000 00b0:
 RIGHT 0x7fd88b407dd0 @ 00b0 00b0: sym=_end
