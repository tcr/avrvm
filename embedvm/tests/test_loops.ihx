:20000000f19040009805a8a6130093aaa604a0060091b19d00918040a0eb0091b29d009168
:2000200081c54090ada4f50041009814a8a61a0091804000928490aaa604a0ef01c591805e
:20004000410092b39da0e400c59180409832a8a610009817aaa604a0080091b49da0ea00a4
:200060009381410001800192b59d0001809832aaa604a019000180928490aaa604a008004a
:20008000018091b69d01918041a0da9040009805a8a61e00928490aba604a00a009092b7d7
:1000a0009d90a607009192b79d00918040a0e09c92
:00000001ff
