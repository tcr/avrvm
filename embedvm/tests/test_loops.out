Called user function 2 with 1 args: 0
Called user function 5 with 2 args: 1 3
Called user function 6 with 1 args: 3
Called user function 5 with 2 args: 2 4
Called user function 6 with 1 args: 4
Called user function 5 with 2 args: 3 5
Called user function 6 with 1 args: 5
Called user function 5 with 2 args: 4 6
Called user function 6 with 1 args: 6
Called user function 5 with 2 args: 5 7
Called user function 6 with 1 args: 7
Called user function 5 with 2 args: 6 8
Called user function 6 with 1 args: 8
Called user function 5 with 2 args: 7 9
Called user function 6 with 1 args: 9
Called user function 5 with 2 args: 8 10
Called user function 6 with 1 args: 10
Called user function 5 with 2 args: 9 11
Called user function 6 with 1 args: 11
Called user function 5 with 2 args: 10 12
Called user function 6 with 1 args: 12
Called user function 5 with 2 args: 11 13
Called user function 6 with 1 args: 13
Called user function 5 with 2 args: 12 14
Called user function 6 with 1 args: 14
Called user function 5 with 2 args: 13 15
Called user function 6 with 1 args: 15
Called user function 5 with 2 args: 14 16
Called user function 6 with 1 args: 16
Called user function 5 with 2 args: 15 17
Called user function 6 with 1 args: 17
Called user function 5 with 2 args: 16 18
Called user function 6 with 1 args: 18
Called user function 5 with 2 args: 17 19
Called user function 6 with 1 args: 19
Called user function 5 with 2 args: 18 20
Called user function 6 with 1 args: 20
Called user function 5 with 2 args: 19 21
Called user function 6 with 1 args: 21
Called user function 5 with 2 args: 20 22
Called user function 6 with 1 args: 22
Called user function 5 with 2 args: 21 23
Called user function 6 with 1 args: 23
Called user function 5 with 2 args: 22 24
Called user function 6 with 1 args: 24
Called user function 5 with 2 args: 23 25
Called user function 6 with 1 args: 25
Called user function 5 with 2 args: 24 26
Called user function 6 with 1 args: 26
Called user function 5 with 2 args: 25 27
Called user function 6 with 1 args: 27
Called user function 5 with 2 args: 26 28
Called user function 6 with 1 args: 28
Called user function 5 with 2 args: 27 29
Called user function 6 with 1 args: 29
Called user function 5 with 2 args: 28 30
Called user function 6 with 1 args: 30
Called user function 5 with 2 args: 29 31
Called user function 6 with 1 args: 31
Called user function 5 with 2 args: 30 32
Called user function 6 with 1 args: 32
Called user function 5 with 2 args: 31 33
Called user function 6 with 1 args: 33
Called user function 5 with 2 args: 32 34
Called user function 6 with 1 args: 34
Called user function 5 with 2 args: 33 35
Called user function 6 with 1 args: 35
Called user function 5 with 2 args: 34 36
Called user function 6 with 1 args: 36
Called user function 5 with 2 args: 35 37
Called user function 6 with 1 args: 37
Called user function 5 with 2 args: 36 38
Called user function 6 with 1 args: 38
Called user function 5 with 2 args: 37 39
Called user function 6 with 1 args: 39
Called user function 5 with 2 args: 38 40
Called user function 6 with 1 args: 40
Called user function 5 with 2 args: 39 41
Called user function 6 with 1 args: 41
Called user function 5 with 2 args: 40 42
Called user function 6 with 1 args: 42
Called user function 5 with 2 args: 41 43
Called user function 6 with 1 args: 43
Called user function 5 with 2 args: 42 44
Called user function 6 with 1 args: 44
Called user function 5 with 2 args: 43 45
Called user function 6 with 1 args: 45
Called user function 5 with 2 args: 44 46
Called user function 6 with 1 args: 46
Called user function 5 with 2 args: 45 47
Called user function 6 with 1 args: 47
Called user function 5 with 2 args: 46 48
Called user function 6 with 1 args: 48
Called user function 5 with 2 args: 47 49
Called user function 6 with 1 args: 49
Called user function 5 with 2 args: 48 50
Main function returned => Terminating.
